### ISSUE TITLE
MsalArgumentException: Authority passed to silent parameters does not match with the cloud associated to the account

### ISSUE THREAD

**Issue by slowcar** [AzureAD/microsoft-authentication-library-for-android#1402]
**Describe the bug**
We get this error in our logs from time to time. The application works correctly again afterwards.

We only have a minimal configuration file, and do not have a custom authority in it. Redacted client_id and redirect_uri:
```
{
  "client_id" : xxx",
  "authorization_user_agent" : "DEFAULT",
  "redirect_uri" : "msauth://xxx",
  "account_mode" : "MULTIPLE",
  "broker_redirect_uri_registered": true
}
```
`Authority authority = publicClientApplicationConfiguration.getDefaultAuthority();`

**Smartphone (please complete the following information):**
 - Device: Different devices, e.g. Pixel 3, Huawei NEM-L51
 - Android Version: Different versions, e.g. 7.0, 11.0
 - MSAL Version 2.0.8

**Stacktrace**
SilentTokenCommandParameters:authorityMatchesAccountEnvironment -  [2021-05-18 13:31:45 - {"thread_id":"29566","correlation_id":"2969f273-c420-429e-b047-a1b6916a7497","x-client-Ver":"1.5.9","x-client-SKU":"MSAL.Android"}] Unable to perform cloud discovery Android 30
     Caused by: com.microsoft.identity.common.exception.ArgumentException: Authority passed to silent parameters does not match with the cloud associated to the account.
        at com.microsoft.identity.common.internal.commands.parameters.SilentTokenCommandParameters.validate(SourceFile:5)
        at com.microsoft.identity.common.internal.controllers.LocalMSALController.acquireTokenSilent(SourceFile:7)
        at com.microsoft.identity.common.internal.commands.SilentTokenCommand.execute(SourceFile:8)
        at com.microsoft.identity.common.internal.commands.SilentTokenCommand.execute(SourceFile:1)
        at com.microsoft.identity.common.internal.controllers.CommandDispatcher.executeCommand(SourceFile:1)
        at com.microsoft.identity.common.internal.controllers.CommandDispatcher.access$200(SourceFile:1)
        at com.microsoft.identity.common.internal.controllers.CommandDispatcher$1.run(SourceFile:10)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:923)

**To Reproduce**
I am unable to trigger this behavior, it just happens from time to time, on my device about 1/week

**Expected behavior**
MSAL works consistently

**Actual Behavior**
The faulty behavior is resolved with the next call to acquireTokenSilent

**Additional context**
https://github.com/AzureAD/microsoft-authentication-library-for-android/issues/1249 seems related, but the resolution "The missing property for authorities was the rootcause" in unclear to me / does not help me, as we use the default configuration.

**Comment by rpdome** [comment]
Hi @slowcar 

The error message here is misleading. What you ran into is actually a network issue. (MSAL performed a 'cloud discovery', which is a network operation, and it failed).  

We're planning to rework this layer for the upcoming feature. cc: @iamgusain

**Comment by iambmelt** [comment]
Fixed via https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1440

**Comment by iambmelt** [comment]
Fix shipped in Android-Common `3.4.5`
